### PR TITLE
Two funding rates taker UI

### PR DESCRIPTION
**Pull request by da-kami**
image

image

### CODE REVIEW

**Comment by da-kami** [comment]
fun fact: https://github.com/itchysats/itchysats/issues/1580 is visible here, see the payout of the short position ;)

**Comment by da-kami** [comment]
The "funding rate" display and behavior will definitely require some more documentation in the UI. 
It is not clear how it behaves from just seeing the values. It is also not clear that is actually can be - or + and fluctuates over time. Furthermore, we should somehow explain why there are two different ones for long and short. 
With @bonomat we thought about calling it interest rate instead, but that will obviously not explain the details ;)

File: taker-frontend/src/App.tsx [hunk 1]
```diff
@@ -181,7 +154,7 @@ export const App = () => {
             <Nav
                 walletInfo={walletInfo}
                 connectedToMaker={connectedToMaker}
-                fundingRate={globalTradeParams?.fundingRateHourly}
+                fundingRate={undefined}
```
Comment (thomaseizinger) [inline taker-frontend/src/App.tsx]: Why not just delete the property?

**Review by thomaseizinger — APPROVED** [review]
Nice!

Commits should be squashed IMO.

**Comment by thomaseizinger** [comment]
> The "funding rate" display and behavior will definitely require some more documentation in the UI.
> It is not clear how it behaves from just seeing the values. It is also not clear that is actually can be - or + and fluctuates over time. Furthermore, we should somehow explain why there are two different ones for long and short.
> With @bonomat we thought about calling it interest rate instead, but that will obviously not explain the details ;)

A combined position would give you a natural place to display the funding rate!

**Review by klochowicz — APPROVED** [review]
it looks actually quite natural there! 👍

**Comment by klochowicz** [comment]
> The "funding rate" display and behavior will definitely require some more documentation in the UI. It is not clear how it behaves from just seeing the values. It is also not clear that is actually can be - or + and fluctuates over time. Furthermore, we should somehow explain why there are two different ones for long and short. With @bonomat we thought about calling it interest rate instead, but that will obviously not explain the details ;)

I like the name "interest rate", as this would be consistent with bitmex vocabulary. This is something we could write in our glossary or FAQ for the users (or at least via the tooltip!)

**Comment by thomaseizinger** [comment]
> > The "funding rate" display and behavior will definitely require some more documentation in the UI. It is not clear how it behaves from just seeing the values. It is also not clear that is actually can be - or + and fluctuates over time. Furthermore, we should somehow explain why there are two different ones for long and short. With @bonomat we thought about calling it interest rate instead, but that will obviously not explain the details ;)
> 
> I like the name "interest rate", as this would be consistent with bitmex vocabulary. This is something we could write in our glossary or FAQ for the users (or at least via the tooltip!)

Semantically as far as I understand. CFDs deal with **two** interest rates: one for the base currency and one for the quote currency. The combination of both yields the "funding rate".
Thus it would not be correct to call it "interest rate" because it would not be clear, which one it is.

File: taker-frontend/src/App.tsx [hunk 2]
```diff
@@ -69,40 +67,26 @@ export const App = () => {
...
-                openingFee: order.opening_fee || 0,
-                fundingRateAnnualized: order.funding_rate_annualized_percent || "0",
-                fundingRateHourly: order
-                    ? Number.parseFloat(order.funding_rate_hourly_percent).toFixed(5)
-                    : "0",
-                liquidationPrice: parseOptionalNumber(order.liquidation_price),
+                id: offer.id,
+                initialFundingFeePerParcel: offer.initial_funding_fee_per_parcel,
+                marginPerParcel: offer.margin_per_parcel,
+                price: Number.parseFloat(offer.price),
```
Comment (bonomat) [inline taker-frontend/src/App.tsx]: not particular related to your PR: 

why are `offer.price`, `offer.min_/max_quantity` strings in the first place? Couldn't we just say these are numbers and avoid `Number.parseFloat`?

Comment (da-kami) [reply taker-frontend/src/App.tsx]: I think we could do that - we are probably a bit too "correct" there. In the daemon they are represented as `Decimal`, and we send them out as string. We *could* use a `Decimal` library in the UI code as well, but don't at the moment, so for now we could just just make them numbers I support :)

Comment (da-kami) [reply taker-frontend/src/App.tsx]: I gave this a try, please have another look

Comment (klochowicz) [reply taker-frontend/src/App.tsx]: I think it was a historical decision to control some of the precision-handling in the projection (which exists solely for the purpose of the UI)...

**Review by klochowicz — APPROVED** [review]
LGTM!

File: taker-frontend/src/App.tsx [hunk 3]
```diff
@@ -69,40 +67,26 @@ export const App = () => {
...
-                fundingRateHourly: order
-                    ? Number.parseFloat(order.funding_rate_hourly_percent).toFixed(5)
-                    : "0",
-                liquidationPrice: parseOptionalNumber(order.liquidation_price),
+                id: offer.id,
+                initialFundingFeePerParcel: offer.initial_funding_fee_per_parcel,
+                marginPerParcel: offer.margin_per_parcel,
+                price: offer.price,
+                liquidationPrice: offer.liquidation_price,
+                fundingRateAnnualized: offer?.funding_rate_annualized_percent,
```
Comment (bonomat) [inline taker-frontend/src/App.tsx]: Nit: `?` looks outdated. 

do we still need this function then? :)

**Review by bonomat — APPROVED** [review]
Looks good

**Pull request merged**